### PR TITLE
Avoid unhelpful resolution messages

### PR DESCRIPTION
**Pull request by reinsch82**
Addresses
- #24464
- #24466
- #24467
- #24468

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy [hunk 1]
```diff
@@ -208,7 +208,7 @@ class TestingIntegrationTest extends JUnitMultiVersionIntegrationSpec implements
 
         where:
         framework   | dependency                | superClass
-        "useJUnit"  | "junit:junit:4.13"        | "org.junit.runner.Result"
+        "useJUnit"  | "$testJunitCoordinates"        | "org.junit.runner.Result"
```
Comment (donat) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy:211]: 🧐 indentation

```suggestion
        "useJUnit"  | "$testJunitCoordinates"   | "org.junit.runner.Result"
```

File: subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java [hunk 2]
```diff
@@ -245,27 +246,26 @@ WorkerRequirement getWorkerRequirement(Class<?> executionClass, WorkerSpec confi
...
-        classes.add(actionClass);
-        for (Object param : params) {
-            if (param != null) {
-                classes.add(param.getClass());
-            }
+        if (parameters != null) {
+            return new Class<?>[]{actionClass, parameters.getClass()};
         }
-        return classes.toArray(new Class<?>[0]);
+        return new Class<?>[]{actionClass};
```
Comment (donat) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java:252]: 🎉 for making this more readable.

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**